ports/ryu/portfile.cmake

```
@@ -30,33 +30,59 @@ set(ENV{BAZEL_BIN_PATH} ${BAZEL})
+        endif ()
+        set(BAZEL_COMPILER --compiler=mingw-gcc)
+        # BAZEL_SH can be propagated to the build environment using VCPKG_KEEP_ENV_VARS
+        if (NOT DEFINED ENV{BAZEL_SH})
+            message("BAZEL_SH is not specified, trying to guess...")
+            get_filename_component(DIR ${VCPKG_DETECTED_CMAKE_C_COMPILER} DIRECTORY)
+            # Bazel expects Mingw-w64 to be installed in MSYS2 (pacman -S mingw-w64-x86_64-toolchain).
+            # From BAZEL_SH it finds MSYS2 root, adds "mingw64" to the root and uses this path as the location of Mingw-w64.
+            # It is also possible to use non-MSYS2 binaries with Bazel if they are installed to a directory
+            # whose name ends with "mingw64", such as c:\mingw64 or c:\TDM-GCC-64\mingw64.
+            string(REGEX REPLACE "/mingw64/bin$" "" MSYS2_ROOT ${DIR})
+            string(REPLACE / \\ MSYS2_ROOT ${MSYS2_ROOT})
+            set(ENV{BAZEL_SH} ${MSYS2_ROOT}\\usr\\bin\\bash.exe)
+            message("BAZEL_SH $ENV{BAZEL_SH}")
+        endif ()
+    else ()
+        message(FATAL_ERROR "${TARGET_TRIPLET} is not supported!")
+    endif ()
     if (VCPKG_DETECTED_CMAKE_SYSTEM_PROCESSOR STREQUAL x86)
         set(BAZEL_CPU --cpu=x64_x86_windows)
     elseif (VCPKG_DETECTED_CMAKE_SYSTEM_PROCESSOR STREQUAL ARM)
         set(BAZEL_CPU --cpu=x64_arm_windows)
     elseif (VCPKG_DETECTED_CMAKE_SYSTEM_PROCESSOR STREQUAL ARM64)
-        set(BAZEL_CPU --cpu=x64_arm64_windows)
+        set(BAZEL_CPU --cpu=arm64_windows)
     endif ()
-endif ()
-
-if (VCPKG_DETECTED_CMAKE_HOST_SYSTEM_NAME STREQUAL Darwin)
-    set(ENV{SDKROOT} ${VCPKG_DETECTED_CMAKE_OSX_SYSROOT})
+else ()
+    set(ENV{CC} ${VCPKG_DETECTED_CMAKE_C_COMPILER})
 endif ()
 
 prepare_bazel_opts(VCPKG_COMBINED_C_FLAGS_RELEASE CONLY_OPTS_RELEASE --conlyopt)
 prepare_bazel_opts(VCPKG_COMBINED_C_FLAGS_DEBUG CONLY_OPTS_DEBUG --conlyopt)
 prepare_bazel_opts(VCPKG_COMBINED_STATIC_LINKER_FLAGS_RELEASE LINK_OPTS_RELEASE --linkopt)
 prepare_bazel_opts(VCPKG_COMBINED_STATIC_LINKER_FLAGS_DEBUG LINK_OPTS_DEBUG --linkopt)
 
+if (VCPKG_HOST_IS_OSX)
+    set(ENV{SDKROOT} ${VCPKG_DETECTED_CMAKE_OSX_SYSROOT})
+    set(LINK_OPTS_RELEASE "${LINK_OPTS_RELEASE};--linkopt=-L${VCPKG_DETECTED_CMAKE_OSX_SYSROOT}/usr/lib")
+    set(LINK_OPTS_DEBUG "${LINK_OPTS_DEBUG};--linkopt=-L${VCPKG_DETECTED_CMAKE_OSX_SYSROOT}/usr/lib")
+endif ()
+
 vcpkg_execute_build_process(
-        COMMAND ${BAZEL} --batch build ${BAZEL_CPU} ${CONLY_OPTS_RELEASE} ${LINK_OPTS_RELEASE} --verbose_failures --strategy=CppCompile=standalone //ryu //ryu:ryu_printf
+        COMMAND ${BAZEL} --batch build ${BAZEL_COMPILER} ${BAZEL_CPU} ${CONLY_OPTS_RELEASE} ${LINK_OPTS_RELEASE} --verbose_failures --strategy=CppCompile=standalone //ryu //ryu:ryu_printf
         WORKING_DIRECTORY ${SOURCE_PATH}
         LOGNAME build-${TARGET_TRIPLET}-rel
 )
 
-if (CMAKE_HOST_WIN32)
+if (CMAKE_STATIC_LIBRARY_SUFFIX STREQUAL .lib)
```

I'm torn on whether this should be:

```suggestion
if ("${CMAKE_STATIC_LIBRARY_SUFFIX}" STREQUAL ".lib")
```

I think what you originally wrote is OK though...

---

I believe that all the double quotes in the code are unnecessary, so I removed them. In my opinion, the lines with or without double quotes have the same meaning. However, if you prefer to keep the quotes for readability, it is acceptable.

---

> I believe that all the double quotes in the code are unnecessary, so I removed them (where it possible). In my opinion, the lines with or without double quotes have the same meaning. However, if you prefer to keep the quotes for readability, it is acceptable.

In all cases where the variable is referred to by `${}`, the double quotes do have meaning. Without them, spaces in the original variable are replaced with `;`s (they are interpreted as different parameters to the CMake function in question).

The only case that is borderline is one like this where the variable is referred to by name rather than with `${}`.

---

Apart from space:
All commands are equal, especially `if` commands. Here, it is not necessary to turn variable names (`CMAKE_STATIC_LIBRARY_SUFFIX` is okay) into quoted strings, but you should turn all expressions and literals into quoted strings.
- When some `${foo}` evaluates to an empty string, `if(foo STREQUAL ${foo})` becomes invalid `if(foo STREQUAL)`.
- We had cases where unexpectedly a variable named `1` was defined to mean `0` (or the other way?), giving unexpected results on `if( ... EQUAL ...)`. So any literal which can be a variable name must be quoted (`"1"`, `"0"`).

So I would really apply quotes for `.lib` here, unambiguously marking it as a plain literal expression.

---

Thank you for your advice! I don't fully agree with it, but as I mentioned earlier, using or not using double quotes doesn't make a difference to me. Therefore, I've added double quotes where it is possible. In my personal opinion, not using quotes was better.

---

You didn't need to add quotes on things that don't name variables. Do you want me to just fix that up for you?